Sources/Amplitude/Amplitude.swift

```diff
@@ -267,22 +262,22 @@ public class Amplitude {
             logger?.log(message: "Skip event based on opt out configuration")
             return
         }
-        event.timestamp = event.timestamp ?? Int64(NSDate().timeIntervalSince1970 * 1000)
-        timeline.process(event: event)
+        timeline.process(event: event, inForeground: inForeground)
     }
 
     func onEnterForeground(timestamp: Int64) {
-        _inForeground = true
+        inForeground = true
         let dummySessionStartEvent = BaseEvent(
             timestamp: timestamp,
             sessionId: -1,
             eventType: Constants.AMP_SESSION_START_EVENT
         )
-        timeline.process(event: dummySessionStartEvent)
+        timeline.process(event: dummySessionStartEvent, inForeground: false)
     }
 
-    func onExitForeground() {
-        _inForeground = false
+    func onExitForeground(timestamp: Int64) {
+        inForeground = false
+        timeline.lastEventTime = timestamp
```

Comment:
As I mentioned in the PR description that logic was added from Amplitude-Android.

Between last Amplitude event and onExitForeground can be minutes and hours.
I think, for users/customers a session is a time period when they do something with the app regardless of amplitude events are tracked or not. So session_end in onExitForeground and session_start in onEnterForegroung looks more correct, I think.

Reply:
This behavior makes sense to the mobile. However, there is no Foreground and Background in the browser, so the last event timestamp will determine if we will end a session. I feel this will lead to different SDKs are not aligned.  WDYT? @justin-fiedler

Reply:
Maybe we should auto-refresh current session (update last event time) in browsers on keyboard/mouse/touch events and/or provide public `refreshSession()` method to let call it by developers explicitly.

Reply:
I think this update makes sense and it matches the current [docs about session functionality on mobile vs web](https://help.amplitude.com/hc/en-us/articles/115002323627-Track-sessions-in-Amplitude#how-amplitude-defines-sessions).

For Browser SDK if customer want to track user entry/exit, I think we can recommend they setup their own "app entered" and "app exited" events to extend the session as desired.